kinesis/checkpointers.py

```
@@ -329,13 +334,15 @@ async def _checkpoint(self, shard_id, sequence):
             previous_val = json.loads(previous_val) if previous_val else None
 
             if not previous_val:
-                raise NotImplementedError(
+                raise CheckpointError(
```

Ruff (0.15.10)

[error] Undefined name `CheckpointError`

(F821)